Feature import wrapper function for dnpImport

tkeller12 opened the pull request.

tkeller12 committed:
Adding wrapper function for importing data

Added dnpIO.py with wrapper function for importing different spectrometer data.

tkeller12 committed:
Unittest for wrapper import function

Added unittest for import wrapper function

tkeller12 committed:
Reformatted to PEP8

ylin00 commented on dnplab/__init__.py:
```diff
@@ -7,4 +7,5 @@
 from . import hydrationGUI
 from . import core
 from . import dnpResults
+from .dnpIO import import_data
```
How about `load_data`, `load` per easyspin style, or `read_data` or `read` per `nmrglue.bruker.read`, `pandas.read_XXX` ?

tkeller12 replied:
I like calling the wrapper function "load".

ylin00 commented on dnplab/dnpIO.py:
```diff
@@ -0,0 +1,37 @@
+
+    elif data_type == "vnmrj":
+        return dnpImport.vnmrj.import_vnmrj(path, *args, **kwargs)
+
```
missing `else`

tkeller12 replied:
Added `else`

ylin00 commented on dnplab/dnpIO.py:
```diff
@@ -0,0 +1,37 @@
+    if data_type == None:
+        data_type = autodetect(path)
+
+    if data_type == NotImplemented:
```
NotImplemented data type should raise an Exception?

ylin00 commented on dnplab/dnpIO.py:
```diff
@@ -0,0 +1,37 @@
+
+    Args:
+        path (str): Path to data directory or file
+        data_type (str): Type of data to import
```
How about `manufacturer`? or something else more **specific** and **explicit**

ylin00 replied:
What are the list of possible options?

ylin00 commented on unittests/test_import_wrapper.py:
```diff
@@ -0,0 +1,27 @@
+        self.vnmrj_dir = "./data/vnmrj/10mM_tempol_in_water_array.fid"
+
+    def test_topspin(self):
+        data = import_data(self.topspin_dir, "topspin", 1)  # last arg is expNum
```
`expNum` should be a keyword argument, instead of positional argument, to be explicit and specific.

tkeller12 replied:
Fixed this

tkeller12 committed:
Update dnpIO.py

Changed name of function from "import_data" to "load".
Added "else" with exception at end of if-statements
Raising error if data_type is invalid

tkeller12 committed:
Update test_import_wrapper.py

Updated name of import wrapper

tkeller12 committed:
Reformatting to PEP8

tkeller12 committed:
Fixed __init__.py

Changed function name in __init__.py
Updated docstring in load

thcasey3 reviewed (changes requested):
@tkeller12 @ylin00 It seems to me that the way this is currently arranged the user interface is dnpIO and this calls dnpImport in the background, shouldn't it be the other way around? It will make more sense to a user to use dnpImport than dnpIO. Also, weren't we discussing changing to "load" or "read" rather than "import"? I vote for dnpLoad

tkeller12 committed:
Changing dnpImport to dnpIO

tkeller12 committed:
Adding list of currently supported types for import wrapper

List of data_types for import wrapper
Updated test_topspin

tkeller12 committed:
Merge branch 'develop' into tim-feature-import_wrapper
